igmpproxy: fix creation of firewall rules

Mushoz opened the pull request:
The init script for igmpproxy uses the option 'network' both as an interface name for fetching the l3_device name and for creating the firewall rules. This only works if the name of the network and firewall zone are identical.

This commit introduces a new option 'zone' for configuring the upstream and downstream firewall zones in order for the init script to create the required firewall rules automatically. When no such options are given, the init script falls back to not creating the firewall rules and the user can opt to create these manually.

Also deleted the phyint name from the example config, because it might cause confusion what this value should be set to, even though this doesn't do anything.

ldir-EDB0 commented:
LGTM - fell over this problem myself recently.

ldir-EDB0 commented on package/network/services/igmpproxy/files/igmpproxy.init:
```diff
@@ -79,18 +79,20 @@ igmp_add_firewall_routing() {
 	json_add_object ""
 	json_add_string type rule
-	json_add_string src "$network"
+	json_add_string src "$zone"
```
I think you need to add 'json_add_string family ipv4' in this section, otherwise ipv6 igmp rules get created.  Since igmp is ipv4 only (AFAIK) it makes little sense to add to the ipv6 firewall.

Mushoz replied:
The rules that were already in place were not created by me. However, I believe you are right in that igmp is ipv4 only, and hence I've added your suggestion to the init script to reflect this as well.

ldir-EDB0 commented:
I think the current situation is best as it gives the option of not auto creating the rules.

Mushoz commented:
This has been my first pull request for OpenWRT/Lede. Do you happen to know how the reviewing/merging process looks like? What does it take for the changes to get merged in? :)

bobafetthotmail commented:
It may take a lot of time as people with commit access can review this when they have time. Months of waiting are normal.

I think you should place the description you wrote in the OP in the actual commit, they are not going to accept it without a description in the commit.

ldir-EDB0 commented:
Yes a concise & accurate description of the commit & why it is required should be in the git commit message.  Also looking back over this, you need to squash the firewall ipv4 change into one single commit.

Mushoz committed:
igmpproxy: fix creation of firewall rules

The init sccript for igmpproxy uses the option 'network' both as an interface name for fetching the l3_device name and for creating the firewall rules. This only works if the name of the network and firewall zone are identical.

This commit introduces a new option 'zone' for configuring the upstream and downstream firewall zones in order for the init script to create the required firewall rules automatically. When no such options are given, the init script falls back to not creating the firewall rules and the user can opt to create these manually.

Signed-off-by: Jaap Buurman <jaapbuurman@gmail.com>

Mushoz commented:
Done. If anything else needs to happen / can be improved, please let me know :)

dedeckeh commented:
Patch pushed to master in commit https://github.com/openwrt/openwrt/commit/0b04926433b60ac1ec448f9dfa8a36dfe5bdcae9; thx
